librz/core/canalysis.c

```diff
@@ -7126,3 +7126,135 @@ RZ_API st64 rz_core_analysis_calls_count(RZ_NONNULL RzCore *core) {
 	}
 	return cov;
 }
+
+static const char *RzCoreAnalysisNameTypeStrs[] = {
+	"var",
+	"function",
+	"flag",
+	"address",
+};
+
+/**
+ * \brief Convert \p typ to string (const char*)
+ */
+RZ_API RZ_BORROW const char *rz_core_analysis_name_type_to_str(RzCoreAnalysisNameType typ) {
+	switch (typ) {
+	case RZ_CORE_ANALYSIS_NAME_TYPE_VAR:
+	case RZ_CORE_ANALYSIS_NAME_TYPE_FUNCTION:
+	case RZ_CORE_ANALYSIS_NAME_TYPE_FLAG:
+	case RZ_CORE_ANALYSIS_NAME_TYPE_ADDRESS:
+		return RzCoreAnalysisNameTypeStrs[typ];
+	default:
+		rz_warn_if_reached();
+		return NULL;
+	}
+}
+
+RZ_API void rz_core_analysis_name_free(RZ_NULLABLE RzCoreAnalysisName *p) {
+	if (!p) {
+		return;
+	}
+	free(p->name);
+	free(p->realname);
+	free(p);
+}
+
+/**
+ * \brief Rename whatever var/flag/function is used at \p addr to \p name
+ * \return success?
+ */
+RZ_API bool rz_core_analysis_rename(RZ_NONNULL RzCore *core, RZ_NONNULL const char *name, ut64 addr) {
+	rz_return_val_if_fail(core && core->analysis && RZ_STR_ISNOTEMPTY(name), false);
+
+	ut64 oldoff = core->offset;
+	if (addr != core->offset) {
+		rz_core_seek(core, addr, true);
+	}
+
```

Comment:
Why is this required?

Reply:
I see that we always use `core->block` as a global buffer, then only the data of `core->block[core->offset..core->offset+core->blocksize]` is valid, so even if it is given an `addr` argument but it does not guarantee that the data at the `addr` is valid.

Reply:
Forget it when I didn't say